sympy/core/tests/test_relational.py

```diff
@@ -1129,3 +1132,54 @@ def test_multivariate_linear_function_simplification():
 
 def test_nonpolymonial_relations():
     assert Eq(cos(x), 0).simplify() == Eq(cos(x), 0)
+
+def test_18778():
+    raises(TypeError, lambda: is_le(Basic(), Basic()))
+    raises(TypeError, lambda: is_gt(Basic(), Basic()))
+    raises(TypeError, lambda: is_ge(Basic(), Basic()))
+    raises(TypeError, lambda: is_lt(Basic(), Basic()))
+
+def test_EvalEq():
+    from sympy import Expr
+
+    class PowTest(Expr):
+        def __new__(cls, base, exp):
+           return Basic.__new__(PowTest, _sympify(base), _sympify(exp))
+
+
+        def _eval_Eq(lhs, rhs):
```

Comment:
There should only one blank line above.

This test should have a comment clarifying that it is testing `_eval_Eq` for backwards compatibility because the new method is to use `_eval_is_eq`